### ISSUE TITLE
Problem with zbassmusic.dll

### ISSUE THREAD

**Issue by Rafiss92** [Silver-Ore-Team/zBassMusic#2]
Hi,
I have a problem with running your plugin. I can't run it on a clean installation of Gothic II NotR. Stepps I did before:
1. Installation of G2NotR (Polish CD edition)
2. G2MDK 2.6
3. G2MDK 2.6a
4. Playerkit + UNION 1.0m (with Resource Manager)
5. I copied my modification files to _work\data (scripts\System\Musicinst.d I left unchanged)
6. I copied zBassMusic.vdf to Gothic II\Data
7. I copied zBassMusic.dll, bass.dll, UnionAPI.dll to Gothic II\System
8. I did reparse scripts and got an Access Violation like that:
![AccessViolation](https://github.com/Silver-Ore-Team/zBassMusic/assets/102917447/9c8029b8-8265-454e-945c-41c2812a0673)
The fun fact is that I could run that on my second laptop and it works perfectly. My friends also have an AV error like that or the plugin just doesn't work for them.
Maybe I did something wrong?

**Comment by Rafiss92** [comment]
I found out why it won't work. I had to install plugin Marvin Helper. Could you explain why is it needed to work with your plugin and why it is not mentioned in your readme?

**Comment by piotrmacha** [comment]
Marvin Helper is not required for the plugin - it should be standalone, but the order of .dll loading could be an issue because `UnionAPI.dll` should load first before other .dll. To verify the order you can enable Union Console in `SystemPack.ini`:
```
[CORE]
ShowDebugWindow = true
```
Then you can check if `UnionAPI.dll` is loaded first.

The Console also has zBassMusic logs that may contain other useful information.

### Possible fix: load UnionAPI.dll first
Inside `SystemPack.ini` (or your mod .ini file) you can specify plugin list to force `UnionAPI.dll` first:
```
[PLUGINS]
PluginList = *UnionAPI
```

We'll try to replicate this Access Violation and update docs with Troubleshooting section when we gather root causes of possible errors.

**Comment by Rafiss92** [comment]
You're right, zBassMusic.vdf is loading before Union.vdf but it loads automatically from `data/plugins`.

![not_works_but_loaded](https://github.com/Silver-Ore-Team/zBassMusic/assets/102917447/20097ed0-4c57-4dc4-9f3b-c0af6b28f789)
![not_works_but_loaded_bottom](https://github.com/Silver-Ore-Team/zBassMusic/assets/102917447/67a2680a-de77-41b6-8a13-7c4182b96c50)

I don't know how to change priority of vdf preloading at the start but still when I tried to set it up with `systempack.ini` it still doesn't worked when I removed vdfs from `data\plugins`. I just left the union.vdf in `data` folder and wrote in `systempack.ini`. All dlls are in `system`. My `system\autorun` is empty.
```ini
[PLUGINS]
PluginList = UnionAPI.dll, zBassMusic.dll, bass.dll
UseVDFS = true
```
I tried to change UseVDFS to false but It doesn't worked too.

Screenshots with removed `marvinhelper` (caused Access Violation):
![works1](https://github.com/Silver-Ore-Team/zBassMusic/assets/102917447/8ddc7a80-fcd7-4c92-87e5-d5629f4a5abf)
![works2](https://github.com/Silver-Ore-Team/zBassMusic/assets/102917447/21a5ca3b-648e-4da2-ba99-68245c9d3f73)

**Comment by muczc1wek** [comment]
@Rafiss92 Your problem have been fixed in the [`0.0.19`](https://github.com/Silver-Ore-Team/zBassMusic/releases/tag/v0.0.19) . Try now and see if it works for you.

**Comment by muczc1wek** [comment]
@Rafiss92 Your problem have been fixed in the [`0.0.19`](https://github.com/Silver-Ore-Team/zBassMusic/releases/tag/v0.0.19) . Try now and see if it works for you.
